src/main/java/org/folio/rest/impl/RequestObjectFactory.java

```diff
@@ -22,27 +24,32 @@
   private final VertxOkapiHttpClient httpClient;
   private final ItemRepository itemRepository;
   private final UserRepository userRepository;
+  private final HoldingsRecordRepository holdingsRecordRepository;
 
   RequestObjectFactory(VertxOkapiHttpClient httpClient, Map<String, String> okapiHeaders) {
     this.okapiHeaders = okapiHeaders;
     this.httpClient = httpClient;
     this.itemRepository = new ItemRepository(httpClient);
     this.userRepository = new UserRepository(httpClient);
+    this.holdingsRecordRepository = new HoldingsRecordRepository(httpClient);
   }
 
   CompletableFuture<JsonObject> createRequestByItem(String patronId, String itemId, Hold entity) {
 
     return completedFuture(new RequestContext(patronId, itemId, entity))
       .thenCompose(this::fetchItem)
+      .thenCompose(this::fetchInstanceId)
       .thenCompose(this::fetchUser)
       .thenCompose(this::fetchRequestType)
       .thenApply(context -> {
         if (context.getRequestType() != RequestType.NONE) {
           final JsonObject holdJSON = new JsonObject()
             .put("requestLevel", "Item")
             .put("requestType", "Hold")
+            .put("instanceId", context.getInstanceId())
             .put(Constants.JSON_FIELD_ITEM_ID, itemId)
-            .put("holdingsRecordId", context.getItem().getString(Constants.JSON_FIELD_HOLDINGS_RECORD_ID))
+            .put(Constants.JSON_FIELD_HOLDINGS_RECORD_ID,
+              context.getItem().getString(Constants.JSON_FIELD_HOLDINGS_RECORD_ID))
```

Comment:
can item be null?

Reply:
Item actually can't be null, cause in this case ItemRepository is going to return Not found Exception